usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/ValidateRouteLongNameDoesNotContainShortName.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2020. MobilityData IO.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.usecase;
+
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.routes.Route;
+import org.mobilitydata.gtfsvalidator.domain.entity.notice.error.RouteLongNameEqualsShortNameNotice;
+import org.mobilitydata.gtfsvalidator.domain.entity.notice.error.RouteShortNameTooLongNotice;
+import org.mobilitydata.gtfsvalidator.usecase.port.GtfsDataRepository;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+
+import java.util.Collection;
+
+/**
+ * Use case to validate that a Route long name does not equal or contain the short name.
+ */
+public class ValidateRouteLongNameDoesNotContainShortName {
+
+    private final GtfsDataRepository dataRepo;
+    private final ValidationResultRepository resultRepo;
+
+    /**
+     * @param dataRepo   a repository storing the data of a GTFS dataset
+     * @param resultRepo a repository storing information about the validation process
+     */
+    public ValidateRouteLongNameDoesNotContainShortName(final GtfsDataRepository dataRepo,
+                                                        final ValidationResultRepository resultRepo) {
+        this.dataRepo = dataRepo;
+        this.resultRepo = resultRepo;
+    }
+
+    /**
+     * Use case execution method: checks if a Route long name does equal or contain the short name
+     * for every Routes in a {@link GtfsDataRepository}. If both are equals, a new error notice is generated.
+     * If long name contains short name, a warning error notice is generated.
+     * This notice is then added to the {@link ValidationResultRepository} provided in the constructor.
+     */
+    public void execute() {
+        Collection<Route> routes = dataRepo.getRouteAll();
+        routes.stream()
+                .filter(route -> route.getRouteLongName().contains(route.getRouteShortName()))
+                .forEach(route -> {
+                    if (route.getRouteLongName().equals(route.getRouteShortName())) {
```

Comment:
Related to the above - I'm not sure this logic is right, but I also don't see any unit tests to confirm it.

It seems like this:
* Filters for all routes where long name contains short name
* For remaining routes
    * if they are equal add `RouteLongNameEqualsShortNameNotice` notice
    * Otherwise add a `RouteShortNameTooLongNotice`

I believe the last bullet should actually be:
* Otherwise add a `RouteLongNameContainsShortNameNotice`

Reply:
Indeed, it's changed now. Unit tests are added as well.